### ISSUE TITLE
[ADAP-631] [Dynamic Tables] Add DOWNSTREAM option for TARGET_LAG

### ISSUE THREAD

**Issue by Fleid** [dbt-labs/dbt-snowflake#666]
The `TARGET_LAG` user interface parameter should be set to the value of `DOWNSTREAM` in addition to `'num seconds | minutes | hours | days'`.

The complete definition becomes:

___

### `TARGET_LAG = { num { seconds | minutes | hours | days } | DOWNSTREAM }`
Specifies the lag for the dynamic table:

#### `'num seconds | minutes | hours | days'`
Specifies the maximum amount of time that the dynamic table’s content should lag behind updates to the base tables.

For example:

If the data in the dynamic table should lag by no more than 5 minutes, specify 5 minutes.

If the data in the dynamic table should lag by no more than 5 hours, specify 5 hours.

The minimum value is 1 minute. If the dynamic table A depends on another dynamic table B, the minimum lag for A must be greater than or equal to the lag for B.

#### `DOWNSTREAM`
Specifies that dynamic table should be refreshed if any downstream ancestor is refreshed.

**Comment by McKnight-42** [comment]
https://github.com/dbt-labs/dbt-snowflake/issues/734

**Comment by noel** [comment]
I know this is to allow for `DOWNSTREAM`, but is `seconds` also a problem?

```
  002755 (22023): SQL compilation error: Invalid TARGET_LAG value '10 seconds'. Dynamic Tables do not support lag values under 60 second(s).
````

**Comment by dataders** [comment]
closed by #734
